### PR TITLE
MAYA-114336 MayaUSD : Optimizations and fixes to tests input/output f…

### PR DESCRIPTION
**Pull request by boudrey**
…olders

Avoid copying test files and scripts by managing input and output(tmpdir) separately.
Make sure build of target "install" is empty after build.
Fix CMake warnings in tests.

### CODE REVIEW

File: test/testSamples/CMakeLists.txt [hunk 1]
```diff
@@ -1,11 +1 @@
...
-add_custom_target(${TARGET_NAME} ALL)
-
-# Copy all of the resources in this directory to a
-# "test/testSamples" subdirectory under the top-level build directory.
-mayaUsd_copyDirectory(${TARGET_NAME}
-    SOURCE ${CMAKE_CURRENT_SOURCE_DIR}
-    DESTINATION ${CMAKE_BINARY_DIR}/test/testSamples
-    EXCLUDE "*.txt"
-)
+# testSamples
```
Comment (seando-adsk) [inline test/testSamples/CMakeLists.txt:1]: Do we even need this CMakeLists.txt file anymore? Could we remove it and then remove the add_subdirectory(testSamples) up one level?

File: test/testUtils/CMakeLists.txt [hunk 2]
```diff
@@ -1,13 +1 @@
...
-# Copy all of the resources and Python scripts in this directory to a
-# "test/python" subdirectory under the top-level build directory. This path
-# will be added to PYTHONPATH for tests so that these utilities are available
-# to all tests.
-mayaUsd_copyDirectory(${TARGET_NAME}
-    SOURCE ${CMAKE_CURRENT_SOURCE_DIR}
-    DESTINATION ${CMAKE_BINARY_DIR}/test/python
-    EXCLUDE "*.txt"
-)
+# This path will be added to PYTHONPATH for tests so that these utilities are available to all tests.
```
[the remark this answers is not in the document]
Comment (seando-adsk) [reply test/testUtils/CMakeLists.txt:1]: So basically you removed the copy of these python files to the build folder and instead added to PYTHONPATH the directory in the source tree?

**Review by seando-adsk — APPROVED** [review]

File: test/testUtils/CMakeLists.txt (hunk 2 above)
[the remark this answers is not in the document]
Comment (boudrey) [reply test/testUtils/CMakeLists.txt:1]: Yes, the .pyc are written in the source tree. They are already excluded from git. I think the time saving is worth it.

**Review by seando-adsk — APPROVED** [review]
Looks good, thanks.

**Pull request merged**